File: src/jsonrpc/debug/methods.js
/**
 * Debug JSON-RPC Methods
 *
 * This module provides a type-safe mapping of debug namespace methods to their types.
 * All imports are kept separate to maintain tree-shakability.
 */

// Method imports - each import is separate for tree-shaking
import * as debug_getBadBlocks from "./getBadBlocks/debug_getBadBlocks.js";
import * as debug_getRawBlock from "./getRawBlock/debug_getRawBlock.js";
import * as debug_getRawHeader from "./getRawHeader/debug_getRawHeader.js";
import * as debug_getRawReceipts from "./getRawReceipts/debug_getRawReceipts.js";
import * as debug_getRawTransaction from "./getRawTransaction/debug_getRawTransaction.js";

/**
 * Method name enum - provides string literals for each method
 *
 * @typedef {(typeof DebugMethod)[keyof typeof DebugMethod]} DebugMethod
 */
export const DebugMethod = {
	debug_getBadBlocks: "debug_getBadBlocks",
	debug_getRawBlock: "debug_getRawBlock",
	debug_getRawHeader: "debug_getRawHeader",
	debug_getRawReceipts: "debug_getRawReceipts",
	debug_getRawTransaction: "debug_getRawTransaction",
};

// Re-export individual method modules for direct access (tree-shakable)
export {
	debug_getBadBlocks,
	debug_getRawBlock,
	debug_getRawHeader,
	debug_getRawReceipts,
	debug_getRawTransaction,
};

File: src/jsonrpc/debug/methods.go
package debug

// Package debug provides Debug JSON-RPC methods.
//
// This file provides a type-safe mapping of debug namespace methods.

// Method name constants
const (
	MethodDebugGetBadBlocks = "debug_getBadBlocks"
	MethodDebugGetRawBlock = "debug_getRawBlock"
	MethodDebugGetRawHeader = "debug_getRawHeader"
	MethodDebugGetRawReceipts = "debug_getRawReceipts"
	MethodDebugGetRawTransaction = "debug_getRawTransaction"
)

// MethodRegistry maps method names to their string identifiers
var MethodRegistry = map[string]string{
	"debug_getBadBlocks": MethodDebugGetBadBlocks,
	"debug_getRawBlock": MethodDebugGetRawBlock,
	"debug_getRawHeader": MethodDebugGetRawHeader,
	"debug_getRawReceipts": MethodDebugGetRawReceipts,
	"debug_getRawTransaction": MethodDebugGetRawTransaction,
}
